File: src/app/frontend/music-vanilla/src/Components/NowPlaying/NowPlaying.ts
import placeholderAlbumCover from '../../Assets/Images/300.png';
import './now-playing.css';
import { Song } from "../../Types/Song.type";

// Mock data for the currently playing song
const currentSong: Song = {
    songTitle: "Song Title",
    artistName: "Artist",
    albumName: "Album",
    albumCoverUrl: placeholderAlbumCover
};

// Root (prefix) for the component's styles
const styleRoot = "now-playing-component";

export function NowPlaying(): HTMLElement {
    const nowPlayingComponent = document.createElement('div');
    nowPlayingComponent.classList.add(styleRoot);
    nowPlayingComponent.classList.add('styled-container');

    const albumArt = document.createElement('img');
    albumArt.src = currentSong.albumCoverUrl;
    albumArt.alt = "Album Art";
    nowPlayingComponent.appendChild(albumArt);

    const contentContainer = document.createElement('div');
    contentContainer.className = `${styleRoot}-text-container`;

    const headerTitle = document.createElement('h1');
    headerTitle.textContent = "Mario's Now Playing";
    contentContainer.appendChild(headerTitle);

    const musicInfo = document.createElement('div');
    musicInfo.className = `${styleRoot}-text`;

    const songTitle = document.createElement('h2');
    songTitle.textContent = currentSong.songTitle;
    musicInfo.appendChild(songTitle);

    const artistName = document.createElement('p');
    artistName.textContent = currentSong.artistName;
    musicInfo.appendChild(artistName);

    const albumName = document.createElement('p');
    albumName.textContent = currentSong.albumName;
    musicInfo.appendChild(albumName);

    // Assemble the content container
    contentContainer.appendChild(musicInfo);

    // Append the content container to the modal
    nowPlayingComponent.appendChild(contentContainer);

    return nowPlayingComponent;
}

File: src/app/frontend/music/music-leptos/src/components/now_playing.rs
use leptos::prelude::*;
use stylance::*;

use crate::models::song::Song;

import_style!(style, "../styles/now_playing.module.scss");

// Define your component
#[component]
pub fn NowPlaying() -> impl IntoView {
    // Mock data for the currently playing song
    let current_song = Song {
        song_title: "Song Title".into(),
        artist_name: "Artist".into(),
        album_name: "Album".into(),
        album_cover_url: "/images/300.png".into(),
    };

    view! {
        <div class=(style::nowPlayingComponent, style::styledContainer).join_classes() >

            // PRODUCTION
            <img src={"/leptos".to_owned() + &current_song.album_cover_url} alt="Album Art" />
            // DEVELOPMENT
            // <img src={&current_song.album_cover_url} alt="Album Art" />

            <div class=style::nowPlayingComponentTextContainer>
                <h1>"Mario's Now Playing"</h1>
                <div class=style::nowPlayingComponentText>
                    <h2>{current_song.song_title}</h2>
                    <p>{current_song.artist_name}</p>
                    <p>{current_song.album_name}</p>
                </div>
            </div>
        </div>
    }
}
